x-pack/plugin/fleet/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/fleet/30_secrets_post.yml

```diff
@@ -0,0 +1,8 @@
+"Create Fleet Secret":
```

Comment:
for the yml tests, we might want to add additional tests with a role that does not have the `read_fleet_secret`/`write_fleet_secret` privileges.

Reply:
is there an example how to set up a role that doesn't have those privileges?

Reply:
yes, so there's 2 approaches we can take here:

A) https://github.com/elastic/elasticsearch/pull/97853 - the Elasticsearch users are created via API in yml test files themselves - we just need to take care to delete them in the teardown section

B) or we can define a `roles.yml` like in https://github.com/elastic/elasticsearch/pull/94011 - this means there's no need to do always setup the users and roles in the setup section of each file. the roles and the defined users can be reused in yml files.
For this we need to add a roles.yml file - e.g. this is how you would define an unprivileged role:

https://github.com/elastic/elasticsearch/blob/faa004df2f1f42688656a14f972d8f1f7deb4b0e/x-pack/plugin/ent-search/qa/rest/roles.yml#L28-L35

then we need to change build.gradle:
https://github.com/elastic/elasticsearch/blob/faa004df2f1f42688656a14f972d8f1f7deb4b0e/x-pack/plugin/ent-search/qa/rest/build.gradle#L14-L22

and then this is how we would test with the user that has the `unprivileged` role:

https://github.com/elastic/elasticsearch/blob/faa004df2f1f42688656a14f972d8f1f7deb4b0e/x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/40_search_application_delete.yml#L24-L37

I think either one is fine, although the roles.yml approach is more prevalent in xpack plugins.

Reply:
I'm also trying to add this, and running into problems, see commit https://github.com/elastic/elasticsearch/pull/97728/commits/1137a5d06936fcd9db4bc075927b23f903c908ab

>    headers: { Authorization: "Basic ZW50c2VhcmNoLXVucHJpdmlsZWdlZDplbnRzZWFyY2gtdW5wcml2aWxlZ2VkLXVzZXI=" }  # unprivileged 

Where is this authorization value coming from? I found the [transform_and_set](https://github.com/elastic/elasticsearch/blob/main/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/README.asciidoc#transform_and_set-1) feature, but I'm getting an error when trying to use it.

```
java.lang.RuntimeException: Failure at [fleet/40_secrets_get:35]: Cannot invoke "org.elasticsearch.test.rest.yaml.ClientYamlTestResponse.evaluate(String, org.elasticsearch.test.rest.Stash)" because "this.response" is null
```